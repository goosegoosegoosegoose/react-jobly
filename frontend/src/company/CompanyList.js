import React, {useState, useEffect} from "react";
import JoblyApi from "../api";
import CompanyCard from "./CompanyCard";
import SearchForm from "../common/SearchForm";

const CompanyList = () => {
  const [companies, setCompanies] = useState([]);

  useEffect(function fetchCompaniesWhenMounted() {
    async function fetchCompanies(){
      const res = await JoblyApi.request("companies");
      setCompanies(res.companies);
    }
    fetchCompanies();
  }, []);

  const getCompanies = (results) => {
    setCompanies([...results.companies]);
  };

  if (companies.length === 0) {
    return (
      <div>
        <SearchForm type="companies" header="name" get={getCompanies} />
        <p>Sorry, no results were found!</p>
      </div>
    )
  };

  return (
    <div>
      <SearchForm type="companies" header="name" get={getCompanies} />
      {companies.map(company => 
        <CompanyCard
          key={company.handle}
          handle={company.handle}
          name={company.name}
          description={company.description}
        />
      )}
    </div>
  )
}

export default CompanyList;