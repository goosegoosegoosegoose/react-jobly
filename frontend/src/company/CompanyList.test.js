import { render } from '@testing-library/react';
import CompanyList from './CompanyList';
import { MemoryRouter } from 'react-router-dom';

test('smoke test', () => {
  render(
    <MemoryRouter>
      <CompanyList />
    </MemoryRouter>
  );
});

test('snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyList />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});