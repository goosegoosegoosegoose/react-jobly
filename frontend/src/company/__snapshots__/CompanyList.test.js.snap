// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        placeholder="Enter search term.."
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <p>
      Sorry, no results were found!
    </p>
  </div>
</DocumentFragment>
`;
